gluten-core/src/main/java/io/glutenproject/vectorized/NativeExpressionEvaluator.java

```diff
@@ -85,6 +87,13 @@ public GeneralOutIterator createKernelWithBatchIterator(
       Plan wsPlan, List<GeneralInIterator> iterList, List<Attribute> outAttrs)
       throws RuntimeException, IOException {
     long allocId = NativeMemoryAllocators.contextInstance().getNativeInstanceId();
+    // Add all static conf to session conf
+    // TODO what if static conf gets changed?
+    final Map<String, String> conf = new HashMap<>(GlutenConfig.getNativeSessionConf());
+    conf.putAll(
```

Comment:
I would suggest add this to `BackendsApiManager.getSettings().getBackendConfigPrefix()` to `GlutenConfig.getNativeSessionConf()`

Reply:
So all config keys with a BE prefix will be both static and dynamic at the same time.

I am feeling that we need to do some refactoring on this part of code to distinguish more clearly on "static" and "dynamic" config. E.g. static should literally be static, changing it will cause an error. Then we list the static config keys one by one in a container source file.

Opening a ticket for that: 
https://github.com/oap-project/gluten/issues/1283

I used to looking into this issue for a while but didn't end up with a solution. Maybe anyone is willing to help or when I get some free time.